roles/parse_operator_metadata/tasks/main.yml

```diff
@@ -1,109 +1,147 @@
-- name: "Check if operator install mode supports AllNamespaces"
-  shell: "{{ yq_bin_path }} r --tojson {{ csv_path }} 'spec' | {{ jq_bin_path }} '.installModes[] | select(.type==\"AllNamespaces\").supported'"
-  register: operator_allnamespaces_support_result
-
-- name: "Check if operator install mode supports OwnNamespace"
-  shell: "{{ yq_bin_path }} r --tojson {{ csv_path }} 'spec' | {{ jq_bin_path }} '.installModes[] | select(.type==\"OwnNamespace\").supported'"
-  register: operator_ownnamespace_support_result
-
-- name: "Check if operator install mode supports SingleNamespace"
-  shell: "{{ yq_bin_path }} r --tojson {{ csv_path }} 'spec' | {{ jq_bin_path }} '.installModes[] | select(.type==\"SingleNamespace\").supported'"
-  register: operator_singlenamespace_support_result
-
-- name: "Check if operator install mode supports MultiNamespace"
-  shell: "{{ yq_bin_path }} r --tojson {{ csv_path }} 'spec' | {{ jq_bin_path }} '.installModes[] | select(.type==\"MultiNamespace\").supported'"
-  register: operator_multinamespace_support_result
-
-- set_fact:
-    operator_allnamespaces_support: "{{ operator_allnamespaces_support_result.stdout if operator_allnamespaces_support_result.stdout != '' else false }}"
-    operator_ownnamespace_support: "{{ operator_ownnamespace_support_result.stdout if operator_ownnamespace_support_result.stdout != '' else false }}"
-    operator_singlenamespace_support: "{{ operator_singlenamespace_support_result.stdout if operator_singlenamespace_support_result.stdout != '' else false }}"
-    operator_multinamespace_support: "{{ operator_multinamespace_support_result.stdout if operator_multinamespace_support_result.stdout != '' else false }}"
-
-- name: "Output all collected data to a yaml file in work dir"
-  template:
-    src: "parsed_operator_data.yml.j2"
-    dest: "{{ work_dir }}/parsed_operator_data.yml"
+- name: "Running Prepare operator metadata"
+  block:
+
+  - name: "Determine the package path inside operator metadata directory"
+    find:
+      paths: "{{ operator_work_dir }}"
+      file_type: file
+      contains: 'packageName'
+    register: package_path_result 
+
+    # assuming only one file matches the file path
+  - name: "set variable for package_path from package_path_result"
+    set_fact:
+      package_path: "{{ package_path_result.files[0].path }}"
+
+  - name: "Load variables from package path into variable package_vars"
+    include_vars:
+      file: "{{ package_path }}"
+      name: package_vars
+
+  - name: "GET package_name variable from loaded package_vars"
+    set_fact:
+      package_name: "{{ package_vars['packageName'] }}"
+
+  - name: "Set default channel when the current_channel is empty"
+    set_fact:
+      current_channel: "{{ (current_channel |length > 0) |
+                            ternary(current_channel,
+                                    package_vars['defaultChannel'] |
```

Comment:
We also need to check if the defaultChannel isn't an empty string. In case it is, we need to default to the first channel value.